testing/matrix-sdk-integration-testing/src/tests/redaction.rs

```diff
@@ -0,0 +1,120 @@
+    let bg_syncer = tokio::spawn(async move { bg_sync.sync(Default::default()).await });
+
+    let room = tamatoa.create_room(request).await?;
+    bg_syncer.abort();
+    // let's send a specific state event
+
+    let avatar_url = mxc_uri!("mxc://example.org/avatar").to_owned();
+    let content = assign!(RoomAvatarEventContent::new(), {
+        url: Some(avatar_url),
+    });
+
+    room.send_state_event(content, "").await?;
+    // sync up.
+    for _ in 0..2 {
+        // we call sync up to three times to give the server time to flush other
+        // messages over and send us the new state event
+        tamatoa.sync_once(Default::default()).await?;
+        if room.get_state_event(StateEventType::RoomAvatar, "").await?.is_some() {
+            break;
+        }
+    }
+
+    // check state event.
+
+    let raw_event =
+        room.get_state_event(StateEventType::RoomAvatar, "").await?.expect("Room Avatar not found");
+    let room_avatar_event: SyncRoomAvatarEvent = raw_event.deserialize_as()?;
+    assert!(
+        room_avatar_event.as_original().expect("event exists").content.url.is_some(),
+        "Event not found"
+    );
+
+    room.redact(room_avatar_event.event_id(), None, None).await?;
+    // sync up.
+    tamatoa.sync_once(Default::default()).await?;
+
+    let raw_event =
+        room.get_state_event(StateEventType::RoomAvatar, "").await?.expect("Room Avatar not found");
+    let room_avatar_event: SyncRoomAvatarEvent = raw_event.deserialize_as()?;
+    // Avatar content has been redacted
+    assert!(room_avatar_event.as_original().is_none(), "Event still found");
+
+    Ok(())
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn test_redaction_static() -> Result<()> {
+    let tamatoa = get_client_for_user("tamatoa".to_owned()).await?;
+    // create a room
+    let request = assign!(CreateRoomRequest::new(), {
+        is_direct: true,
+    });
+
+    // we need a background sync for `create_room`
+    let bg_sync = tamatoa.clone();
+    let bg_syncer = tokio::spawn(async move { bg_sync.sync(Default::default()).await });
+
+    let room = tamatoa.create_room(request).await?;
+    bg_syncer.abort();
+
+    // let's send a specific state event
+    let avatar_url = mxc_uri!("mxc://example.org/avatar").to_owned();
+    let content = assign!(RoomAvatarEventContent::new(), {
+        url: Some(avatar_url),
+    });
+
+    room.send_state_event(content, "").await?;
+    // sync up.
+    for _ in 0..2 {
+        // we call sync up to three times to give the server time to flush other
+        // messages over and send us the new state event
+        tamatoa.sync_once(Default::default()).await?;
+        if room.get_state_event_static::<RoomAvatarEventContent>("").await?.is_some() {
+            break;
+        }
+    }
+
+    // check state event.
+
+    let room_avatar_event: SyncRoomAvatarEvent =
+        room.get_state_event_static("").await?.expect("Room Avatar not found").deserialize()?;
```

Comment:
Passing an empty string seems like the `get_state_event_static` method should take a `Option<&str>` or something similar, thoughts?

Reply:
Not sure... the spec states that `state_key` is required, but may be an empty string for certain types. And as `"" != None`, I wouldn't agree that this was a better API (as under that format `None` could also be understood as "don't match the specific state_key, just hand out the first that matches the type").

Reply:
We have strong state key types in Ruma. I don't think `Option` makes sense, but we could have separate `get_state_event_static` + `get_state_event_static_with_key` functions where the former works for (most) state events which have [`StateKey = EmptyStateKey`](https://docs.ruma.io/ruma_common/events/room/name/struct.RoomNameEventContent.html#impl-StateEventContent-for-RoomNameEventContent) and the latter works for everything and has stronger types (so getting a member event would require you to pass [a User ID](https://docs.ruma.io/ruma_common/events/room/member/struct.RoomMemberEventContent.html#impl-StateEventContent-for-RoomMemberEventContent) for example).